### PR TITLE
Add posibility to set partitionKey when creating record

### PR DESCRIPTION
**Pull request by bobo**
new PR with squached commits since i reforked the repo and github doesnt pick up changes

### CODE REVIEW

**Comment by jmatusewicz** [comment]
Thanks @bobo for cleaning up the git history. 

You still have a failing test in `RequestFactoryTest`, because now the partition key gets also serialized. You have to tell the jackson object mapper to ignore this attribute during serialization. An easy way would be to put an `@JsonIgnore` annotation on the `partitionKey` parameter. A better way would be to configure a custom serializer for `Record` in `JacksonConfiguration`.

**Comment by bobo** [comment]
added a jsonignore to the field.
sorry about the delay :-(

**Comment by jrehwaldt** [comment]
Since you proposed this change we made a couple of changes which expose the `Record` instances during consumption (check out `KinesisInboundHandler`). From consumers point of view the value for `partionKey` is not correctly set in this case. Therefore, we are unfortunately not able to merge this PR as-is anymore.

**Comment by JoakimLofgren** [comment]
@jrehwaldt would something like this, solve it?

```
± git diff
diff --git a/src/main/kotlin/de/bringmeister/spring/aws/kinesis/ObjectMapperRecordDeserializerFactory.kt b/src/main/kotlin/de/bringmeister/spring/aws/kinesis/ObjectMapperRecordDeserializerFactory.kt
index 3d58f71..e93c909 100644
--- a/src/main/kotlin/de/bringmeister/spring/aws/kinesis/ObjectMapperRecordDeserializerFactory.kt
+++ b/src/main/kotlin/de/bringmeister/spring/aws/kinesis/ObjectMapperRecordDeserializerFactory.kt
@@ -26,7 +26,8 @@ class ObjectMapperRecordDeserializerFactory(
             val json = Charset.forName("UTF-8")
                 .decode(awsRecord.data)
                 .toString()
-            return objectMapper.readValue<Record<D, M>>(json, type)
+            val record = objectMapper.readValue<Record<D, M>>(json, type)
+            return Record(record.data, record.metadata, awsRecord.partitionKey)
         }
     }
 }
```

Did you also want the `partitionKey` exposed in the `KinesisInboundHandler` interface ?

**Comment by jrehwaldt** [comment]
Okay, that'll work. If you update your code accordingly and add a test for this, I'll merge.

**Comment by JoakimLofgren** [comment]
@jrehwaldt Is that good enough?

File: src/main/kotlin/de/bringmeister/spring/aws/kinesis/ObjectMapperRecordDeserializerFactory.kt [hunk 1]
```diff
@@ -26,7 +27,10 @@ class ObjectMapperRecordDeserializerFactory(
             val json = Charset.forName("UTF-8")
                 .decode(awsRecord.data)
                 .toString()
-            return objectMapper.readValue<Record<D, M>>(json, type)
+            val record = objectMapper.readValue<Record<D, M>>(json, type)
+            val partitionKey = awsRecord.partitionKey ?: UUID.randomUUID().toString()
```
[the remark this answers is not in the document]
Comment (JoakimLofgren) [reply src/main/kotlin/de/bringmeister/spring/aws/kinesis/ObjectMapperRecordDeserializerFactory.kt]: Done

**Comment by jrehwaldt** [comment]
Thank you.

**Pull request merged**